code/modules/projectiles/projectile.dm

```diff
@@ -631,4 +631,58 @@ var/list/impact_master = list()
 
 /obj/item/projectile/attack_hand(mob/user)
 	if(timestopped)
-		..()
\ No newline at end of file
+		..()
+
+/obj/item/projectile/friendlyCheck
+	invisibility = 101
+	rotate = 0
+	damage = 0
+	nodamage = 1
+	var/atom/impact = null
+
+/obj/item/projectile/friendlyCheck/process()
+	OnFired()
+	while(!impact && loc && (kill_count > 0))
+		if(dist_x > dist_y)
+			bresenham_step(dist_x,dist_y,dx,dy)
+		else
+			bresenham_step(dist_y,dist_x,dy,dx)
+	return impact
+
+/obj/item/projectile/friendlyCheck/bresenham_step(var/distA, var/distB, var/dA, var/dB)
```

Comment:
What's the point of having this here?
All I can see is that it doesn't die (which can be solved by making kill_count INFINITY) and it doesn't sleep (which can be solved by making step_delay 0).
Also it doesn't return anything for some reason.